modules/javafx.base/src/main/java/com/sun/javafx/collections/VetoableListDecorator.java

```diff
@@ -100,23 +101,34 @@ public void removeListener(InvalidationListener listener) {
         helper = ListListenerHelper.removeListener(helper, listener);
     }
 
+    @Override
+    public final void doSort(Comparator<? super E> comparator) {
+        var sortedList = new ArrayList<>(list);
+        sortedList.sort(comparator);
+        setAllImpl(Collections.unmodifiableList(sortedList));
+    }
+
     @Override
     public boolean addAll(E... elements) {
         return addAll(Arrays.asList(elements));
     }
 
     @Override
     public boolean setAll(E... elements) {
-        return setAll(Arrays.asList(elements));
+        return setAllImpl(new UnmodifiableArrayList<>(elements, elements.length));
     }
 
     @Override
     public boolean setAll(Collection<? extends E> col) {
-        List<E> elements = unmodifiableList(col);
-        onProposedChange(elements, 0, size());
+        return setAllImpl(unmodifiableList(col));
+    }
+
+    private boolean setAllImpl(List<E> unmodifiableList) {
+        onProposedChange(unmodifiableList, 0, size());
```

Comment:
Have you thought of moving all the checks and wrapping into `setAllImpl()`?

Regardless of that, the current code is good.

Reply:
The wrappers are slightly different (`UnmodifiableArrayList` vs `Collections.unmodifiableList`), so doing that would be a little less efficient.